Unable to build http_get with TLS enabled on cc32xx

**Describe the bug**
Build errors are seen when building with TLS enabled:
```

[161/166] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf 
: && ccache /home/local/bin/zephyr-sdk-0.11.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -Wl,-T zephyr/linker.cmd -Wl,-Map=/home/local/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/zephyr_prebuilt.map -Wl,--whole-archive app/libapp.a zephyr/libzephyr.a zephyr/arch/common/libarch__common.a zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a zephyr/lib/libc/newlib/liblib__libc__newlib.a zephyr/lib/posix/liblib__posix.a zephyr/boards/arm/cc3220sf_launchxl/libboards__arm__cc3220sf_launchxl.a zephyr/subsys/net/libsubsys__net.a zephyr/subsys/net/l2/wifi/libsubsys__net__l2__wifi.a zephyr/subsys/net/ip/libsubsys__net__ip.a zephyr/drivers/gpio/libdrivers__gpio.a zephyr/drivers/serial/libdrivers__serial.a modules/ti/simplelink/source/ti/devices/cc32xx/lib..__modules__hal__ti__simplelink__source__ti__devices__cc32xx.a modules/ti/simplelink/lib..__modules__hal__ti__simplelink.a -Wl,--no-whole-archive zephyr/kernel/libkernel.a zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -L"/home/local/bin/zephyr-sdk-0.11.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/thumb/v7e-m/nofp" -L/home/local/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr -lgcc -Wl,--print-memory-usage -mthumb -mcpu=cortex-m4 -Wl,--gc-sections -Wl,--build-id=none -Wl,--sort-common=descending -Wl,--sort-section=alignment -Wl,-u,_OffsetAbsSyms -Wl,-u,_ConfigAbsSyms -nostdlib -static -no-pie -Wl,-X -Wl,-N -Wl,--orphan-handling=warn -mabi=aapcs -march=armv7e-m -lm -lc -lgcc -L"/home/local/bin/zephyr-sdk-0.11.0/arm-zephyr-eabi/arm-zephyr-eabi"/lib/thumb/v7e-m/nofp -specs=nano.specs && :
Memory region         Used Size  Region Size  %age Used
           FLASH:       92764 B         1 MB      8.85%
            SRAM:       25056 B       256 KB      9.56%
        IDT_LIST:         104 B         2 KB      5.08/home/local/bin/zephyr-sdk-0.11.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: app/libapp.a(http_get.c.obj): in function `main':
/home/local/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/../src/http_get.c:65: undefined reference to `tls_credential_add'
/home/local/bin/zephyr-sdk-0.11.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: zephyr/libzephyr.a(simplelink_sockets.c.obj): in function `map_credentials':
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:640: undefined reference to `credential_next_get'
/home/local/bin/zephyr-sdk-0.11.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: /home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:668: undefined reference to `credential_next_get'
collect2: error: ld returned 1 exit status

```
This seems to be due to CONFIG_TLS_CREDENTIALS not being thrown anymore.

**To Reproduce**
Steps to reproduce the behavior:
1. cd zephyr/samples/net/sockets/http_get
2. west build -b cc3220sf_launchxl -p -- -DOVERLAY_CONFIG=overlay-tls-offload.conf
3. See error

**Expected behavior**
The build should succeed.

**Impact**
Prevents use of TLS on cc32xx

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain (e.g Zephyr SDK, ...): SDK 0.11.0
 - Commit SHA or Version used: 9dc4eed622e642e2423e2c71526b95f2b1c1ab08

--- a/subsys/net/lib/sockets/Kconfig
+++ b/subsys/net/lib/sockets/Kconfig
@@ -48,7 +48,7 @@ config NET_SOCKETS_DNS_TIMEOUT
 
 config NET_SOCKETS_SOCKOPT_TLS
 	bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
-	select TLS_CREDENTIALS if NET_NATIVE
+	imply TLS_CREDENTIALS
 	select MBEDTLS if NET_NATIVE
 	help
 	  Enable TLS socket option support which automatically establishes

